File: DACAug2006/C++/OOP/ADT/banktest1.cpp
#include "bank.h"
#include <iostream>

using namespace Banking;
using namespace std;

int main()
{
	Account* cust = new SavingsAccount;
	cust->Deposit(9500);
	Account* vend = new CurrentAccount;
	double amt;
	cout << "Enter amount to transfer : ";
	cin >> amt;
	try
	{
		cust->Transfer(amt, vend);
	}
	catch(InsufficientFunds)
	{
		cout << "Transfer aborted due to lack of funds"
			 << endl;
	}
	cout << "Customer's Account ID is " << cust->ID()
		 << " and Balance is " << cust->Balance()
		 << endl;
	cout << "Vendor's Account ID is " << vend->ID()
		 << " and Balance is " << vend->Balance()
		 << endl;
	
}

File: DACAug2006/Java/Core/Language/BankTest1.java
import banking.*;

class BankTest1{
	
	public static void main(String[] args){
		Banker b = Banker.getBanker();
		Account cust = b.openAccount(4500, true);
		Account vend = b.openAccount(2000);
		try{
			double amt = Double.parseDouble(args[0]);
			cust.transfer(amt, vend);
		}catch(InsufficientFundsException e){
			System.out.println("Transfer aborted due to lack of funds!");
		}catch(NumberFormatException e){
			System.out.println("Bad input!");
		}finally{
			System.out.println("Customer Account ID is " + cust.getId() + " and Balance is " + cust.getBalance());
		}
		System.out.println("Vendor Account ID is " + vend.getId() + " and Balance is " + vend.getBalance());	
	}
}